### DISCUSSION TITLE
Compile using Visual Studio 2019

### DISCUSSION THREAD

**Discussion by kurlyak** [LostArtefacts/TRX#1054] [General]
I trying to compile this project with VS2019 and get this linker error:

Error	LNK2001	unresolved external symbol _av_packet_alloc	TOMBRAID	D:\TOMBRAID\s_audio_stream.obj	1	

This function av_packet_alloc from FFmpeg, I dont know what library file to need.

**Comment by oziphantom** [comment]
you need FFMPEG but then ones that are build for 32bit. I used the ones from here https://github.com/sudo-nautilus/FFmpeg-Builds-Win32/releases `ffmpeg-master-latest-win32-gpl-shared.zip ` 
you will also need to get SDL2 32bit and zlib 32bit.

**Comment by kurlyak** [comment]
Thank you very much. Project is now almost full linked. There no previous linker errors FFmpeg. But there is another linker problem, several functions caused error like this:

Error	LNK2001	unresolved external symbol _Effect_Blood	TOMBRAID	I:\TOMBRAID\teeth_trap.obj	1	

It is function Effect_Blood from TR file blood.h and blood.c - this files are added to VS2019 project's solution. What is the problem?

P.S.

Found answer who need. 

File blood.h and blood.c there are in two copies, first copy in location:

game\objects\effects

second copy in location

\game\effects

And when VS2019 produce blood.obj in Debug directory - cannot be two copy of same file in same directory - when first OBJ created it replaced with second OBJ file. 

I renamed blood2.h and blood2.h one of copies - and it's works.

**Reply by walkawayy** [reply]
@rr- maybe one of the `blood.c`s should be renamed?

**Reply by kurlyak** [reply]
Thank you, I renamed files in project, now works.
